crates/vfs-notify/src/lib.rs

```diff
@@ -119,18 +119,19 @@ impl NotifyActor {
                         self.watched_dir_entries.clear();
                         self.watched_file_entries.clear();
 
-                        let send = |msg| (self.sender)(msg);
-                        send(loader::Message::Progress {
-                            n_total,
-                            n_done: LoadingProgress::Started,
-                            config_version,
-                            dir: None,
-                        });
+                        self.sender
+                            .send(loader::Message::Progress {
+                                n_total,
+                                n_done: LoadingProgress::Started,
+                                config_version,
+                                dir: None,
+                            })
+                            .unwrap();
 
                         let (entry_tx, entry_rx) = unbounded();
                         let (watch_tx, watch_rx) = unbounded();
                         let processed = AtomicUsize::new(0);
-                        config.load.into_par_iter().enumerate().for_each(move |(i, entry)| {
+                        config.load.into_par_iter().enumerate().for_each(|(i, entry)| {
```

Comment:
This is the cause for the deadlock, the `move` moved `entry_tx` and `watch_tx` before and then dropped them once the `for_each` is done. Now they stick around causing the `watch_rx` and `entry_rx` iteration afterwards to not exit as the corresponding senders are not yet dropped

Reply:
Thanks! Fixed :)